src/arch.rs

```diff
@@ -0,0 +1,175 @@
+use crate::{
+    starknet::{ArrayAbi, U256},
+    types::TypeBuilder,
+    values::JitValue,
+};
+use bumpalo::Bump;
+use cairo_lang_sierra::{
+    extensions::{
+        core::{CoreLibfunc, CoreType, CoreTypeConcrete},
+        starknet::{secp256::Secp256PointTypeConcrete, StarkNetTypeConcrete},
+    },
+    ids::ConcreteTypeId,
+    program_registry::ProgramRegistry,
+};
+use std::ptr::NonNull;
+
+mod aarch64;
+mod x86_64;
+
+/// Implemented by all supported argument types.
+pub trait AbiArgument {
+    fn to_bytes(&self, buffer: &mut Vec<u8>);
+}
+
+pub struct JitValueWithInfoWrapper<'a> {
+    pub value: &'a JitValue,
+    pub type_id: &'a ConcreteTypeId,
+    pub info: &'a CoreTypeConcrete,
+
+    pub arena: &'a Bump,
+    pub registry: &'a ProgramRegistry<CoreType, CoreLibfunc>,
+}
+
+impl<'a> JitValueWithInfoWrapper<'a> {
+    fn map<'b>(
```

Comment:
can you add some docs to this function? It's short but I think it can help.

Reply:
Done. Not to the impl but in the trait.